src/main/java/com/atomicjar/logging/LogFmtMarker.java

```diff
@@ -26,13 +26,13 @@ public class LogFmtMarker implements Marker {
 
     private final Map<String, Object> data;
 
-    private LogFmtMarker(Map<String, Object> data) {
-        this.data = Collections.unmodifiableMap(data);
+    private LogFmtMarker(Map<String,Object> data) {
+        this.data = data;
     }
 
 
     public static LogFmtMarker with(String key, Object value) {
-        return new LogFmtMarker(Map.of(key, value));
+        return new LogFmtMarker(Collections.emptyMap()).and(key, value);
```

Comment:
Looking at the code, null value removes the key from the existing map. But with empty map that doesn't make a lot of sense, so you could simply check if it's null and directly return the marker with an empty map.

Reply:
done